smarts/core/route_cache.py

```diff
@@ -69,9 +69,23 @@ def road_ids(self) -> List[str]:
 
     @classmethod
     def from_road_ids(
-        cls, road_map: RoadMap, road_ids: Sequence[str]
+        cls,
+        road_map: RoadMap,
+        road_ids: Sequence[str],
+        resolve_intermediaries: bool = False,
     ) -> "RouteWithCache":
         """Factory to generate a new RouteWithCache from a sequence of road ids."""
+
+        if len(road_ids) > 0 and resolve_intermediaries:
+            routes = road_map.generate_routes(
+                start=road_map.road_by_id(road_ids[0]),
+                end=road_map.road_by_id(road_ids[-1]),
+                via=road_ids[1:-1],
+                max_to_gen=1,
+            )
```

Comment:
+ The `via` should be of type `Sequence[Road]`.
+ The arguments are for `start_road` and `end_road`, instead of `start` and `end`.